### PR TITLE
Add hreflang links to Fronts and Tag Pages

### PR DESCRIPTION
**Pull request by arelra**
## What does this change?

Add [hreflang links](https://developers.google.com/search/docs/specialty/international/localized-versions#language-codes) to network fronts and [editionalised](https://github.com/guardian/dotcom-rendering/blob/814001ade13b0861f196d2452fd6bf4f261426f9/dotcom-rendering/src/lib/edition.ts#L80-L94) pages.

This signals to Google to treat each page as distinct for each language locale.

The links exist on Frontend but were missed in the migration to DCR.

## Why?

SEO

### After

Network front:

```
<link rel="alternate" href="https://m.code.dev-theguardian.com/uk" hreflang="en-GB" />
<link rel="alternate" href="https://m.code.dev-theguardian.com/us" hreflang="en-US" />
<link rel="alternate" href="https://m.code.dev-theguardian.com/au" hreflang="en-AU" />
<link rel="alternate" href="https://m.code.dev-theguardian.com/europe" hreflang="en-EU" />
<link rel="alternate" href="https://m.code.dev-theguardian.com/international" hreflang="en" />
```

Editionalised page:

(europe and international do not have editionalised pages)

```
<link rel="alternate" href="https://m.code.dev-theguardian.com/uk/sport" hreflang="en-GB" />
<link rel="alternate" href="https://m.code.dev-theguardian.com/us/sport" hreflang="en-US" />
<link rel="alternate" href="https://m.code.dev-theguardian.com/au/sport" hreflang="en-AU" />
```

### CODE REVIEW

File: dotcom-rendering/src/layouts/TagPageLayout.tsx [hunk 1]
```diff
@@ -185,7 +185,9 @@ export const TagPageLayout = ({ tagPage, NAV }: Props) => {
 					image={tagPage.header.image}
 				/>
 				{tagPage.groupedTrails.map((groupedTrails, index) => {
-					const locale = getEditionFromId(tagPage.editionId).locale;
+					const locale = getEditionFromId(
+						tagPage.editionId,
+					).dateLocale;
```
Comment (mxdvl) [inline dotcom-rendering/src/layouts/TagPageLayout.tsx:190]: Looking at this, I wonder if destructuring might be both explicit and shorter.

```suggestion
					const { dateLocale } = getEditionFromId(
						tagPage.editionId,
					);
```

File: dotcom-rendering/src/lib/alternate-lang-links.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	}, []);
+};
+
+const createAlternateLangLinks = (
+	baseUrl: string,
+	pageId: string,
+): string[] => {
+	if (isNetworkFront(pageId)) {
+		const pageEdition = getEditionFromPageId(pageId);
+		const hasLangLocale = pageEdition?.langLocale;
```
Comment (mxdvl) [inline dotcom-rendering/src/lib/alternate-lang-links.ts:42]: Maybe we want to make sure this is a `boolean`?

```suggestion
		const hasLangLocale = !!pageEdition?.langLocale;
```

File: dotcom-rendering/src/lib/edition.test.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+describe('is network front', () => {
+	it('returns true if pageId is a network front', () => {
+		expect(everyNetworkFront.every((page) => isNetworkFront(page))).toBe(
+			true,
+		);
+	});
+	it('returns false if pageId is NOT a network front', () => {
+		expect(
+			everyEditionalisedPage.every((page) => isNetworkFront(page)),
```
Comment (mxdvl) [inline dotcom-rendering/src/lib/edition.test.ts:24]: You might find the following more succint and explicit?

```suggestion
			everyEditionalisedPage.every(isNetworkFront),
```

File: dotcom-rendering/src/server/htmlPageTemplate.ts [hunk 4]
```diff
@@ -199,12 +201,13 @@ https://workforus.theguardian.com/careers/product-engineering/
...
                 <title>${title}</title>
                 <meta name="description" content="${he.encode(description)}" />
+				<meta charset="utf-8">
 				${
 					canonicalUrl !== undefined
 						? `<link rel="canonical" href="${canonicalUrl}" />`
 						: '<!-- no canonical URL -->'
 				}
-                <meta charset="utf-8">
+				${alternateLangLinks?.join('\n') ?? ''}
```
Comment (mxdvl) [inline dotcom-rendering/src/server/htmlPageTemplate.ts:210]:
```suggestion
				${alternateLangLinks?.join('\n') ?? '<!-- no alternate langs for the current page -->'}
```

File: dotcom-rendering/src/server/htmlPageTemplate.ts [hunk 5]
```diff
@@ -67,6 +68,7 @@ export const htmlPageTemplate = (props: WebProps | AppProps): string => {
 		renderingTarget,
 		hasPageSkin = false,
 		weAreHiring,
+		alternateLangLinks,
```
Comment (mxdvl) [inline dotcom-rendering/src/server/htmlPageTemplate.ts:71]: Wondering if it’s easier to reason about empty arrays than undefined ones…

```suggestion
		alternateLangLinks = [],
```

File: dotcom-rendering/src/lib/edition.ts [hunk 6]
```diff
@@ -1,63 +1,106 @@
...
 		longTitle: 'Europe edition',
 		title: 'Europe edition',
-		locale: 'en-gb',
+		dateLocale: 'en-gb',
 		timeZone: 'Europe/Paris',
+		hasEditionalisedPages: false,
 	},
+];
+
+const ukEdition = editionList[0];
```
Comment (mxdvl) [inline dotcom-rendering/src/lib/edition.ts:75]: If you’re the destructuring afficionado…

```suggestion
const [ ukEdition ] = editionList;
```

File: dotcom-rendering/src/lib/edition.ts [hunk 7]
```diff
@@ -1,63 +1,106 @@
...
+
+const editionalisedPages = [
```
Comment (mxdvl) [inline dotcom-rendering/src/lib/edition.ts:77]: Would it possibly be worth adding [a link to where these are defined in `frontend`](https://github.com/guardian/frontend/blob/60c6b55944d52d87039a2a844665cf39dc7fe437/common/app/common/Edition.scala#L20-L35) for future travellers?

**Review by mxdvl — COMMENTED** [review]
Love a good `i18n`

File: dotcom-rendering/src/lib/alternate-lang-links.ts (hunk 2 above)
Comment (arelra) [reply dotcom-rendering/src/lib/alternate-lang-links.ts]: Thanks for the spots!

File: dotcom-rendering/src/lib/edition.ts (hunk 7 above)
Comment (mxdvl) [reply dotcom-rendering/src/lib/edition.ts:80]: JSDoc?

**Review by mxdvl — APPROVED** [review]

File: dotcom-rendering/src/server/htmlPageTemplate.ts (hunk 4 above)
Comment (arelra) [reply dotcom-rendering/src/server/htmlPageTemplate.ts]: I think I will leave as is, it seems redundant for the normative case.

**Pull request merged**